src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -147,11 +147,14 @@ internal static void AddMvcCoreServices(IServiceCollection services)
             //
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, MvcCoreMvcOptionsSetup>());
+            services.TryAddEnumerable(
+                ServiceDescriptor.Transient<IPostConfigureOptions<MvcOptions>, MvcOptionsConfigureCompatibilityOptions>());
```

Comment:
Just wanted to call this out regarding `Anything the developer sets explicitly is
an override and always wins.`...if a user wants to override one particular setting(like out of 4 switches for example..remember these 4 switches could have also cascaded across multiple minor releases) that a compatibility version is setting, they would need to do something like(which isn't bad)..i.e use [PostConfigure](https://github.com/aspnet/Options/blob/dev/src/Microsoft.Extensions.Options/IPostConfigureOptions.cs#L8) rather than the usual `Configure`
```c#
services.AddMvc().SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_3);
services.PostConfigure<MvcOptions>(o =>
{
    o.AllowBindingUndefinedValueToEnumType = true;
});

```

Reply:
Nah, this is handled here: https://github.com/aspnet/Mvc/pull/7142/files#diff-c9e246ea8e03a841a1b895c97e72dc07R58

If the user manipulates the value it sets `switch.IsValueSet=true` and we don't overwrite it.

Reply:
Ah, I see. Thanks!